src/amazonS3.ts

```diff
@@ -75,20 +75,51 @@ export class AmazonS3 {
     return object;
   }
 
-  public async listCommonPrefixes(key: string): Promise<string[]> {
+  // Paginates listObjectV2 and returns both Contents and CommonPrefixes
+  public async listAllObjects(key: string): Promise<{ contents: S3.ObjectList; commonPrefixes: string[] }> {
     const prefix = key.startsWith(this.baseKey) ? key : `${this.baseKey}/${key}/`;
-    const objects = await this.s3
-      .listObjectsV2({ Bucket: this.options.bucket ?? BUCKET, Delimiter: '/', Prefix: prefix })
-      .promise();
-    return (objects.CommonPrefixes ?? [])?.map((item) => item.Prefix).filter(isString);
+    const bucket = this.options.bucket ?? BUCKET;
+    let continuationToken;
+    const allContents: S3.ObjectList = [];
+    const allCommonPrefixes: string[] = [];
+
+    // Use maximum iteration to ensure termination
+    const MAX_ITERATIONS = 100;
```

Comment:
so now the limit is 100,000?

Reply:
Yea, it would be 🤷  I just wanted to make sure we didn't get stuck in a loop